pkg/imgutil/snapshotter.go

```diff
@@ -77,9 +80,18 @@ func (rs *remoteSnapshotterOpts) isRemote() bool {
 
 func (rs *remoteSnapshotterOpts) apply(config *pull.Config, ref string) {
 	h := ctdsnapshotters.AppendInfoHandlerWrapper(ref)
+
+	if rs.snapshotter == "soci" {
+		h = socisource.AppendDefaultLabelsHandlerWrapper(
+			"", ctdsnapshotters.AppendInfoHandlerWrapper(ref))
+
+		config.RemoteOpts = append(config.RemoteOpts, containerd.WithPullUnpack)
```

Comment:
WithPullUnpack is also appended when unpackB is true. 
Any side effect for double appending? Maybe manually try pulling with soci with unpackB to see what will happen

https://github.com/containerd/nerdctl/blob/ba23d5d7e5eda4d3de739edcbf99ce633c0433f2/pkg/imgutil/imgutil.go#L229

Reply:
I don't think there's any side effect for double appending WithPullUnpack. 

It behaves as expected when unpackB is manually set to true:
```
[chxgaddy@lima-containerd-dev nerdctl]$ sudo ./_output/nerdctl --snapshotter=soci pull 776719656719.dkr.ecr.us-east-1.amazonaws.com/rabbitmq:latest
INFO[0000] unpackB is true
776719656719.dkr.ecr.us-east-1.amazonaws.com/rabbitmq:latest:                     resolved       |++++++++++++++++++++++++++++++++++++++|
manifest-sha256:308429d6a6b8350702cd5813e3e6ae821df60bbaea6e63860e6a9d4f58a1c180: done           |++++++++++++++++++++++++++++++++++++++|
config-sha256:c959ec1fbc2efee44afe586ea6b96456ea62e5ba9a5fca3a02b174ef564beb5c:   done           |++++++++++++++++++++++++++++++++++++++|
elapsed: 9.0 s                                                                    total:  10.4 K (1.2 KiB/s)
[chxgaddy@lima-containerd-dev nerdctl]$ mount | grep -o fuse.rawBridge
fuse.rawBridge
fuse.rawBridge
fuse.rawBridge
```

It also seems like unpackB may be already set to true when soci is specified for some reason. 

(Output when I didn't manually set unpackB to true:
```
[chxgaddy@lima-containerd-dev nerdctl]$ sudo ./_output/nerdctl --snapshotter=soci pull 776719656719.dkr.ecr.us-east-1.amazonaws.com/rabbitmq:latest
INFO[0000] unpackB is true
776719656719.dkr.ecr.us-east-1.amazonaws.com/rabbitmq:latest:                     resolved       |++++++++++++++++++++++++++++++++++++++|
manifest-sha256:308429d6a6b8350702cd5813e3e6ae821df60bbaea6e63860e6a9d4f58a1c180: done           |++++++++++++++++++++++++++++++++++++++|
config-sha256:c959ec1fbc2efee44afe586ea6b96456ea62e5ba9a5fca3a02b174ef564beb5c:   done           |++++++++++++++++++++++++++++++++++++++|
elapsed: 8.9 s                                                                    total:  10.4 K (1.2 KiB/s)
[chxgaddy@lima-containerd-dev nerdctl]$ mount | grep -o fuse.rawBridge
fuse.rawBridge
fuse.rawBridge
fuse.rawBridge
```
)

So, it may not be necessary to append WithPullUnpack in `snapshotter.go` when soci is specified. I'll look more into this, but it looks like there's no side effect to double appending WithPullUnpack.

Reply:
Good. I suggest covering the scenario in e2e tests to further validate it.

Reply:
@CodeChanning Let's add unpackB=true test cases in run and pull?